src/ElementPortal.js

```diff
@@ -1,79 +1,6 @@
-import React, { PropTypes } from 'react';
-import ReactDOM from 'react-dom';
+import React from 'react';
+import FiberElementPortal from './FiberElementPortal';
+import LegacyElementPortal from './LegacyElementPortal';
 
-const reDataAttr = /^data\-(.+)$/;
-
-export const getNodeData = (node) => {
-  // fallback
-  if (!node.dataset) {
-    const result = {};
-    const attributes = node.attributes;
-    for (let i = 0; i < attributes.length; i++) {
-      const attr = attributes[i];
-      const match = attr.name.match(reDataAttr);
-      if (match) {
-        const key = match[1];
-        result[key] = attr.value;
-      }
-    }
-    return result;
-  }
-
-  return Object.keys(node.dataset)
-    .reduce((result, key) => {
-      result[key] = node.dataset[key];
-      return result;
-    }, {});
-};
-
-const ElementPortal = React.createClass({
-  propTypes: {
-    id: PropTypes.string,
-    selector: PropTypes.string,
-    // Remove styles and classes from node.
-    shouldReset: PropTypes.bool,
-    view: PropTypes.func
-  },
-
-  componentDidMount() {
-    this.renderToNodes();
-  },
-
-  componentDidUpdate() {
-    this.renderToNodes();
-  },
-
-  renderToNodes() {
-    const nodeById = this.props.id && document.getElementById(this.props.id);
-    const nodesById = nodeById ? [nodeById] : [];
-    const nodesBySelector = (this.props.selector && [].slice.call(document.querySelectorAll(this.props.selector))) || [];
-    const nodes = nodesById.concat(nodesBySelector);
-
-    nodes.forEach(node => {
-      if (this.props.shouldReset) {
-        node.className = '';
-        node.removeAttribute('style');
-      }
-
-      const View = this.props.view;
-
-      const children = View ?
-        <View domNode={node} data={getNodeData(node)}/> :
-        React.Children.only(this.props.children);
-
-      // This will become `ReactDOM.unstable_createPortal()` once the new Fiber implementation lands in React:
-      // https://github.com/facebook/react/pull/8386
-      ReactDOM.unstable_renderSubtreeIntoContainer(
-        this,
-        children,
-        node
-      );
-    });
-  },
-
-  render() {
-    return null;
-  }
-});
-
-export default ElementPortal;
+const hasNativePortal = 'createPortal' in React;
+export default hasNativePortal ? FiberElementPortal : LegacyElementPortal;
```

Comment:
Nice 👍!

Minor, but since `src/ElementPortal` only contains the conditional to `Fiber` / `Legacy` now, I'm wondering if we could merge this file with `src/index.js`? I think `src/ElementPortal.js` is a slightly misleading name for this file now, especially since we also have two other files named `FiberElementPortal` and `LegacyElementPortal`. One might think at first glance that we have 3 different types of ElementPortal instead of only 2.

Reply:
> Minor, but since src/ElementPortal only contains the conditional to Fiber / Legacy now, I'm wondering if we could merge this file with src/index.js?

It's imported by `withElementPortal`, so it avoids duping the logic.

> I think src/ElementPortal.js is a slightly misleading name for this file now

We could have a directory `ElementPortal` with `index.js`, `ElementPortalFiber.js` and `ElementPortalLegacy` so people understand there are actually 2 implementations?

Reply:
> It's imported by withElementPortal, so it avoids duping the logic.

Aaah, you're right, missed that!

> We could have a directory ElementPortal with index.js, ElementPortalFiber.js and ElementPortalLegacy so people understand there are actually 2 implementations?

That sounds like a good idea! 👍